File: wordcount.py/mapper.pl
#!/usr/bin/env perl

use strict;
use warnings;

while ( my $line = <> ) {
    $line =~ s/(^\s+|\s+$)//g;
    my @keys = split(m/\s+/, $line);
    foreach my $key ( @keys ) {
        my $value = 1;
        printf( "%s\t%d\n", $key, $value );
    }
}

File: wordcount.py/mapper.R
#!/usr/bin/env Rscript

stdin <- file('stdin', open='r')
while ( length(line <- readLines(stdin, n=1, warn=FALSE)) > 0 ) {
    line <- gsub('(^\\s+|\\s+$)', '', line)
    keys <- unlist(strsplit(line, split='\\s+'))
    value <- 1
    lapply(keys, FUN=function(key,value) cat(paste(key,'\t',value,'\n',sep='')), value=value)
}
close(stdin)
